### ISSUE TITLE
[GRF005] CRS scale - formula is wrong

### ISSUE THREAD

**Issue by atomczak** [buildingSMART/ifc-gherkin-rules#522]
The attached IFC file: [Building-Architecture.zip](https://github.com/user-attachments/files/28789440/Building-Architecture.zip) has the following data:

```
#11=IFCGEOMETRICREPRESENTATIONCONTEXT($,'Model',3,0.001,#7,$);
#12=IFCGEOMETRICREPRESENTATIONSUBCONTEXT('Body','Model',*,*,*,*,#11,$,.MODEL_VIEW.,$);
#13=IFCPROJECT('2Ndyd$OSX7s9A04nc4lyye',#1,'ifc silly sample scene - project','Demystifying IFC with a playful scene using diverse building elements and compositions.',$,$,$,(#11),#14);
#14=IFCUNITASSIGNMENT((#15,#16,#17));
#15=IFCSIUNIT(*,.LENGTHUNIT.,.MILLI.,.METRE.);
#18=IFCPROJECTEDCRS('EPSG:32760','EPSG:32760 - WGS 84 / UTM zone 60S','WGS 84',$,$,$,#19);
#19=IFCSIUNIT(*,.LENGTHUNIT.,$,.METRE.);
#20=IFCMAPCONVERSION(#11,#18,729013.3488297004,9063992.684697364,1.3000000000000018,0.4999999999999999,0.8660254037844387,0.001);
```
So:
- Project (source) in **mililetres**
- Projected CRS (target) in **metres**
- Map conversion scale: **0.001**

However, it results in this warning:
<img width="700" height="400" alt="Image" src="https://github.com/user-attachments/assets/8ae0ec2a-161d-4197-86f1-62747cdb197c" />

because of the rule: [GRF005_CRS-unit-type-differences.feature](https://github.com/buildingSMART/ifc-gherkin-rules/blob/main/features/rules/GRF/GRF005_CRS-unit-type-differences.feature)

Since the scale is defined as:
https://github.com/buildingSMART/ifc-gherkin-rules/blob/e2b27c1d9c2cfd9538cd1f6d41d44bba0f24b070/features/rules/GRF/GRF005_CRS-unit-type-differences.feature#L17

$$
Scale = \frac{source}{target}
$$

Therefore:

$$
Scale = \frac{ProjectUnit}{ProjectedCRSUnit} = \frac{milimetre}{metre} = 0.001
$$

So I believe the warning is wrong, the model should pass the validation.

cc: @janbrouwer

**Comment by civilx64** [comment]
This same comment has been raised a few times already privately from vendors and other implementers.  In reviewing the 4.3 documentation as well as additional supporting documents during development of the rule, it was not explicit whether the factor from project context units to map conversion units should be `Scale` or its reciprocal `1 / Scale`. Example: [IFC4.x-development #533](https://github.com/buildingSMART/IFC4.x-development/issues/533#issuecomment-1253098707).

Therefore we made the decision to leave the logic of GRF005 as-is so that solution providers that have already adjusted their implementations are not stuck chasing a moving target in future releases of the Validation Service.

**Comment by atomczak** [comment]
Thanks @civilx64. Indeed, when I flipped the equation like this and put 1000 as the scale, there are no warnings.

$$
Scale = \frac{ProjectedCRSUnit}{ProjectUnit} = \frac{metre}{milimetre} = 1000
$$

@janbrouwer, didn't we have the issue that some IFC viewers didn't display correctly when the scale was 1000, but all was good with 0.001?

Since updating the standard text will take some time, in the meantime, can we update the warning text to be very clear? It took me and @janbrouwer some time to interpret it.

**Comment by janbrouwer** [comment]
All viewers I know of break on this choice when using my model, linking them here to make sure, but it might also be a problem in my model 😬
https://ifcgref.bk.tudelft.nl/ @amiroo4
https://bim.works/ @rubendel 
https://geo.buildingsmart.nl/ @JoostGevaert @JulesBlm
https://www.ifclite.com/ @louistrue

**Comment by janbrouwer** [comment]
Hi @civilx64,

When I read the thread you referenced, my conclusion is actually the opposite.

To me it reinforces that the scale is a transformation parameter applied from source CRS → target CRS, not just a ratio derived from unit factors.

Following that, the mm → m case results in 0.001, since we’re transforming engineering coordinates into map coordinates?

**Comment by JulesBlm** [comment]
I agree with @janbrouwer and I don't think it's that ambiguous. I read the spec as fixing the direction of the transformation, and then there's only one possible value, 0.001.

IFC4 ADD2 TC1, [IfcMapConversion](https://standards.buildingsmart.org/IFC/RELEASE/IFC4/ADD2_TC1/HTML/schema/ifcrepresentationresource/lexical/ifcmapconversion.htm):

> The map conversion deals with transforming the local engineering coordinate system, often called world coordinate system, into the coordinate reference system of the underlying map.

The transform runs `engineering coord system` → `projected map CRS`, so project units in, CRS units out.

For this casne, `X, Y, Z` go in the transform as project units (mm) and `E, N, H` come out as map units (m).
A point at `X = 1000 mm` should land 1 m east of the origin, only `Scale = 0.001` would do that (`0.001 × 1000 = 1 m`).
So I think all the viewers get this right.

I checked the rule's code and I think I found a mismatch. Its text says:

> The map conversion scale must be **the quotient of the project length units and the target CRS length units**

project units / CRS units = mm / m = 0.001. But [`crs.py`](https://github.com/buildingSMART/ifc-gherkin-rules/blob/main/features/steps/steps/crs.py#L86) does the inverse:

```python
quotient = crs_unit_factor / proj_unit_factor
```

Also `proj_unit_factor` is a confusing variable name, because `proj` is also often used for 'map projection'.

**Comment by aothms** [comment]
Thanks for weighing in everyone. There's actually a formula in the subtype that makes things unambiguous for us.

https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcMapConversionScaled.htm

There we also read that the multiplication of scale transforms us from engineering coordinates into map coordinates and 0.001 is correct

**Comment by atomczak** [comment]
Thanks @aothms

**Comment by civilx64** [comment]
Fixed in v1.0.4.
